From 26bf35bb4338caaefad3b1be4ee2c99d8e618618 Mon Sep 17 00:00:00 2001
From: Tiago Gaspar <tiagogaspar8@gmail.com>
Date: Fri, 1 Sep 2023 18:46:15 +0100
Subject: [PATCH] jool: update package to version 4.1.10

Update jool to version 4.1.10 and remove a no longer needed patch.
There was also a need to backport a patch to fix compile in some archs.

Signed-off-by: Tiago Gaspar <tiagogaspar8@gmail.com>
---
 net/jool/Makefile                             |  8 +++---
 ...000-include-csum_tcpudp_magic-header.patch | 26 +++++++++++++++++++
 .../patches/001-libtool-reduce-version.patch  | 21 ---------------
 ...0-fix-compilation-warning-simple-fix.patch |  6 ++---
 4 files changed, 33 insertions(+), 28 deletions(-)
 create mode 100644 net/jool/patches/000-include-csum_tcpudp_magic-header.patch
 delete mode 100644 net/jool/patches/001-libtool-reduce-version.patch

diff --git a/net/jool/Makefile b/net/jool/Makefile
index fbefb3949d39f..6d5069d81552b 100644
--- a/net/jool/Makefile
+++ b/net/jool/Makefile
@@ -8,7 +8,7 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=jool
-PKG_VERSION:=4.1.8
+PKG_VERSION:=4.1.10
 PKG_RELEASE:=1
 
 PKG_LICENSE:=GPL-2.0-only
@@ -16,8 +16,8 @@ PKG_LICENSE_FILES:=COPYING
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/NICMx/Jool.git
-PKG_SOURCE_VERSION:=6822bdee4ec63467e82d723a3381b3116c1853d9
-PKG_MIRROR_HASH:=6ef000459858a87e206c903828f428d469c18221789cb65fec91a8d822b0178f
+PKG_SOURCE_VERSION:=47334c9124b7a2e3253fb279e6c33acb9c2b09a6
+PKG_MIRROR_HASH:=ef34f68eb911c97016180ed09e852625a3c0fe49f7d7574382a03db75c23b60e
 
 PKG_BUILD_DIR=$(KERNEL_BUILD_DIR)/$(PKG_SOURCE_SUBDIR)
 PKG_BUILD_PARALLEL:=1
@@ -67,7 +67,7 @@ endef
 define Package/jool/Default
   SECTION:=net
   CATEGORY:=Network
-  URL:=https://www.jool.mx
+  URL:=https://nicmx.github.io/Jool/
 endef
 
 define Package/jool/Default/description
diff --git a/net/jool/patches/000-include-csum_tcpudp_magic-header.patch b/net/jool/patches/000-include-csum_tcpudp_magic-header.patch
new file mode 100644
index 0000000000000..4d004b865d2f8
--- /dev/null
+++ b/net/jool/patches/000-include-csum_tcpudp_magic-header.patch
@@ -0,0 +1,26 @@
+From aa511f8856b4a437de005f23c6932948fffd8768 Mon Sep 17 00:00:00 2001
+From: Alberto Leiva Popper <ydahhrk@gmail.com>
+Date: Sat, 1 Jul 2023 09:52:09 -0600
+Subject: [PATCH] Include csum_tcpudp_magic's header
+
+Fixes build in some kernels.
+In fact, I'm not sure why I'm not getting the warning. It looks as
+though I should.
+
+Thanks to Charles Hardin for this patch.
+
+Fixes #407.
+---
+ src/mod/common/rfc7915/6to4.c | 1 +
+ 1 file changed, 1 insertion(+)
+
+--- a/src/mod/common/rfc7915/6to4.c
++++ b/src/mod/common/rfc7915/6to4.c
+@@ -1,6 +1,7 @@
+ #include "mod/common/rfc7915/6to4.h"
+ 
+ #include <linux/inetdevice.h>
++#include <net/ip6_checksum.h>
+ #include <net/udp.h>
+ #include <net/tcp.h>
+ 
diff --git a/net/jool/patches/001-libtool-reduce-version.patch b/net/jool/patches/001-libtool-reduce-version.patch
deleted file mode 100644
index 94f3e7586434f..0000000000000
--- a/net/jool/patches/001-libtool-reduce-version.patch
+++ /dev/null
@@ -1,21 +0,0 @@
-From: Tiago Gaspar <tiagogaspar8@gmail.com>
-Date: Fri, 21 Jan 2022 13:39:30 +0000
-Subject: [PATCH] libtool: Reduce the required version
-
-Current Openwrt's libtool verion is 2.4.2, yet, jool had the requirement
-set to 2.4.6 in source.
-Change that to Openwrt's version.
-
-Signed-off-by: Tiago Gaspar <tiagogaspar8@gmail.com>
-
---- a/configure.ac
-+++ b/configure.ac
-@@ -5,7 +5,7 @@ AC_PREREQ([2.68])
- AC_INIT([Jool], [4.1.8], [jool@nic.mx])
- AC_CONFIG_SRCDIR([src/common/xlat.h])
- AM_INIT_AUTOMAKE([subdir-objects])
--LT_PREREQ([2.4.6])
-+LT_PREREQ([2.4.2])
- 
- # Checks for programs.
- AC_PROG_CC
diff --git a/net/jool/patches/100-fix-compilation-warning-simple-fix.patch b/net/jool/patches/100-fix-compilation-warning-simple-fix.patch
index b695d19bb81f9..dccd70f09bbb9 100644
--- a/net/jool/patches/100-fix-compilation-warning-simple-fix.patch
+++ b/net/jool/patches/100-fix-compilation-warning-simple-fix.patch
@@ -1,6 +1,6 @@
 --- a/src/mod/common/skbuff.c
 +++ b/src/mod/common/skbuff.c
-@@ -109,9 +109,9 @@ static void print_skb_fields(struct sk_buff *skb, unsigned int tabs)
+@@ -109,9 +109,9 @@ static void print_skb_fields(struct sk_b
  	print(tabs, "network_header:%u", skb->network_header);
  	print(tabs, "mac_header:%u", skb->mac_header);
  	print(tabs, "head:%p", skb->head);
@@ -15,7 +15,7 @@
  static int truncated(unsigned int tabs)
 --- a/src/mod/common/xlator.c
 +++ b/src/mod/common/xlator.c
-@@ -875,7 +875,7 @@ void xlator_put(struct xlator *jool)
+@@ -889,7 +889,7 @@ void xlator_put(struct xlator *jool)
  static bool offset_equals(struct instance_entry_usr *offset,
  		struct jool_instance *instance)
  {
@@ -26,7 +26,7 @@
  
 --- a/src/mod/common/nl/instance.c
 +++ b/src/mod/common/nl/instance.c
-@@ -37,7 +37,7 @@ static int serialize_instance(struct xlator *entry, void *arg)
+@@ -37,7 +37,7 @@ static int serialize_instance(struct xla
  	if (!root)
  		return 1;
  
